contracts/pTokenNoGSN.sol

```diff
@@ -79,6 +80,11 @@ contract PTokenNoGSN is
             "Recipient cannot be the token contract address!"
         );
         _mint(recipient, value, userData, operatorData);
+        if (userData.length > 0) {
+            // TODO: check if this is needed, and, eventually, replace .code with a function
+            // require(recipient.code.length > 0, "Recipient is not a contract");
+            try IPReceiver(recipient).receiveUserData(userData) {} catch {}
```

Comment:
Maybe worth adding a note to the reader inside the catch to explain that we don't reverts originating from code outside the contract's purview to cause the entire mint to fail, since that part is all we're concerned about here.

Reply:
Done, please have a look